graphene/src/intersection.rs

```diff
@@ -69,3 +71,284 @@ pub fn get_arbitrary_point_on_path(path: &BezPath) -> Option<Point> {
 		PathSeg::Cubic(cubic) => cubic.p0,
 	})
 }
+
+/// \/                               \/
+/// Bezier Curve Intersection algorithm
+/// \/                               \/
+
+/// each intersection has two curves, which are distinguished between using this enum
+/// TODO: refactor so actual curve data and Origin aren't seperate
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Origin {
+	Alpha,
+	Beta,
+}
+
+impl std::ops::Not for Origin {
+	type Output = Self;
+	fn not(self) -> Self {
+		match self {
+			Origin::Alpha => Origin::Beta,
+			Origin::Beta => Origin::Alpha,
+		}
+	}
+}
+
+pub struct Intersect {
+	pub point: Point,
+	pub t_a: f64,
+	pub t_b: f64,
+	pub a_seg_idx: usize,
+	pub b_seg_idx: usize,
```

Comment:
Expand abbreviations. `time_alpha` and `alpha_segement_id`.

Reply:
(or `index` not `id` in this case, from `idx`)

For context, we encourage full names instead of abbreviations because code readability is very important, and the mental burden of expanding abbreviations is more taxing than reading a few extra characters of code. Even if you end up with pretty long variable or function names, descriptivity makes that worthwhile. Soon I'll find the time to write a style/best practices guide so I don't have to mention this to contributors individually! :)

Reply:
Sounds good! I think I'm gonna hold off on making this change for a little while however, I'm considering doing some refactoring concerning the Intersect structure, so when I do that i'll make the changes.